docroot/modules/custom/mass_migrate/mass_migrate.module

```diff
@@ -0,0 +1,14 @@
+<?php
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function mass_migrate_module_implements_alter(&$implementations, $hook) {
+  // Completely take over entity_hierarchy delete hook.
+  if ($hook == 'entity_delete' && \Drupal::state()->get('entity_hierarchy_disable_writes', FALSE)) {
+    unset($implementations['entity_hierarchy']);
+  }
+}
```

Comment:
FYI we have Hux installed and it has this feature more directly available - https://www.drupal.org/project/hux